airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflowImpl.java

```diff
@@ -0,0 +1,237 @@
+import io.airbyte.workers.temporal.scheduling.activities.JobCreationAndStatusUpdateActivity.AttemptFailureInput;
+import io.airbyte.workers.temporal.scheduling.activities.JobCreationAndStatusUpdateActivity.JobCancelledInput;
+import io.airbyte.workers.temporal.scheduling.activities.JobCreationAndStatusUpdateActivity.JobCreationInput;
+import io.airbyte.workers.temporal.scheduling.activities.JobCreationAndStatusUpdateActivity.JobCreationOutput;
+import io.airbyte.workers.temporal.scheduling.activities.JobCreationAndStatusUpdateActivity.JobFailureInput;
+import io.airbyte.workers.temporal.scheduling.activities.JobCreationAndStatusUpdateActivity.JobSuccessInput;
+import io.airbyte.workers.temporal.scheduling.shared.ActivityConfiguration;
+import io.airbyte.workers.temporal.scheduling.state.WorkflowState;
+import io.airbyte.workers.temporal.scheduling.state.listener.NoopStateListener;
+import io.airbyte.workers.temporal.sync.SyncWorkflow;
+import io.temporal.api.enums.v1.ParentClosePolicy;
+import io.temporal.failure.CanceledFailure;
+import io.temporal.failure.ChildWorkflowFailure;
+import io.temporal.workflow.CancellationScope;
+import io.temporal.workflow.ChildWorkflowOptions;
+import io.temporal.workflow.Workflow;
+import java.time.Duration;
+import java.util.Optional;
+import java.util.UUID;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class ConnectionManagerWorkflowImpl implements ConnectionManagerWorkflow {
+
+  public long NON_RUNNING_JOB_ID;
+  public int NON_RUNNING_ATTEMPT_ID;
+
+  private WorkflowState workflowState = new WorkflowState(UUID.randomUUID(), new NoopStateListener());
+
+  Optional<Long> maybeJobId = Optional.empty();
+  Optional<Integer> maybeAttemptId = Optional.empty();
+
+  private final GenerateInputActivity getSyncInputActivity = Workflow.newActivityStub(GenerateInputActivity.class, ActivityConfiguration.OPTIONS);
+  private final JobCreationAndStatusUpdateActivity jobCreationAndStatusUpdateActivity =
+      Workflow.newActivityStub(JobCreationAndStatusUpdateActivity.class, ActivityConfiguration.OPTIONS);
+  private final ConfigFetchActivity configFetchActivity = Workflow.newActivityStub(ConfigFetchActivity.class, ActivityConfiguration.OPTIONS);
+  private final ConnectionDeletionActivity connectionDeletionActivity =
+      Workflow.newActivityStub(ConnectionDeletionActivity.class, ActivityConfiguration.OPTIONS);
+
+  private CancellationScope syncWorkflowCancellationScope;
+
+  public ConnectionManagerWorkflowImpl() {}
+
+  @Override
+  public void run(final ConnectionUpdaterInput connectionUpdaterInput) throws RetryableException {
+    try {
+      if (connectionUpdaterInput.getWorkflowState() != null) {
+        workflowState = connectionUpdaterInput.getWorkflowState();
+      }
+      try {
+        syncWorkflowCancellationScope = Workflow.newCancellationScope(() -> {
+          // Scheduling
+          final ScheduleRetrieverInput scheduleRetrieverInput = new ScheduleRetrieverInput(
```

Comment:
can we refactor this? this is somewhat tough to read

Reply:
I will try after lunch. Let me know if there is on specific pain point.

Reply:
Thanks! I wrote out a suggestion, it was a combination of trailing right brackets and too many things happening in one big method.